### ISSUE TITLE
Cut a release for GHC 8.4

### ISSUE THREAD

**Issue by bgamari** [haskell/haddock#737]
The GHC 8.4.1 release is quickly approaching and i would like to have all of the submodules finalized by next alpha. For this we'll need a new release.

**Comment by alexbiehl** [comment]
I'll take care of that this week.

Ben Gamari <notifications@github.com> schrieb am So., 21. Jan. 2018 um
21:48 Uhr:

**Comment by hvr** [comment]
@bgamari we're currently a bit blocked on https://phabricator.haskell.org/D4336

**Comment by alexbiehl** [comment]
@hvr Do you agree with these versions?

- haddock-2.19.1
- haddock-api-2.19.1
- haddock-library-1.5.1 (we added `Table` constructor to Markup and added a new module)

@bgamari Before cutting, is it possible to merge the following diffs in ghc-8.4?

1) Haddock needs to pass visible modules for instance filtering (https://phabricator.haskell.org/D4290)
2) Support constructor Haddocks in more places (https://phabricator.haskell.org/D4292)
3) Option for LINE pragmas to get lexed into tokens (https://phabricator.haskell.org/D4336)

1) touches a function which is only used by haddock. Impact should be low for ghc. 2) and 3) are a bit more invasive but fix some major annoyances in haddock. They really add vaue and it would be very nice to have it fixed by haddock-2.19.1.

**Comment by bgamari** [comment]
> @bgamari Before cutting, is it possible to merge the following diffs into ghc-8.4?

On it.

**Comment by bgamari** [comment]
At this point I've merged all of the above into master but I'm afraid we'll need to backport the `haddock` patches to the `ghc-8.4` branch. I have a first cut at this in bgamari/haddock@ghc-8.4, but I didn't quite finish yesterday and will be occupied by other things this weekend.

**Comment by bgamari** [comment]
Any word on a cherry-pick of the patches in question?

**Comment by alexbiehl** [comment]
@bgamari 

I have updated `ghc-8.4` branch. Please bump `utils/haddock` to d314d0c3c9d6213417954f46757c511864077927. I excluded Alecs constructor documentation stuff as it seemed difficult to rebase given the `ConDecl` changes.

We need the following patches in ghc-8.4.1:
- https://github.com/ghc/ghc/commit/66961dc85e39bbc43a1c065ee00c381a7ee579e1
- https://github.com/ghc/ghc/commit/9a57cfebd2e65109884712a27a0f29d1a71f57b7

Thank you!

**Comment by hvr** [comment]
@bgamari Once GHC's  ghc-8.4 branch is synced up w/ Haddock, I'd like to give to try it out the ghc-8.4+haddock-2.19 combo to see if I can spot any remaining showblockers.

**Comment by alexbiehl** [comment]
@bgamari I had to bump the branch once again. I forgot about @harpocrates patch to make use of the new lexer feature. Please bump haddock to ac33472e834d381f95fd56586e57e6653263055c.
